pitch.md

```diff
@@ -14,31 +23,170 @@ _What need is met by your idea? Who are you helping? What is that person's
+ideas and think that I would have a lot of fun doing that for the next few
+weeks.
+
+That's not all though. I also generally have a lot of trouble using computers
+to foster creativity. I'm a very paper and pencil sort of guy. I think
+designing a system which has to potential to make computers a better platform
+for exploring ideas (even in the limited domain of classical constructions)
+would be worth putting time into.
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain? Be careful
 to distinguish between, e.g,. a graphical interface and a linguistic interface._ 
 
+#### User Language
+
+The User Language is the way in which users write constructions.
+
+Taken from my project ideas writeup:
+
+> [Users] would write statements like:
+>
+> ```
+> circle x with center C
+> line y tangent to x at point D
+> line z through points C and D
+> ```
+>
+> I really like language idea because it mirrors the way that mathematicians
+> write about geometry. Who hasn't seen something like:
+>
+> Let x be a circle centered at C. Let y be a line tangent to x at point D. Let z
+> be a line through C and D. ...more proof...
+>
+> =============================================================================
+>
+> The more interesting would be interactive mode, where as the users enter each
+> line the image is drawn for them to see what they have, as well as to indicate
+> potentially interesting thing to do next (IE, highlighting the tangent point
+> after drawing a tangent). Using this information the users can issue special
+> interactive commands that act on those possibilities.
+>
+> At the end of the session the user could export the image, LaTeX, or DSL code
+
+#### Geometric Representation
+
+We can also talk about the syntax of the internal geometric representation.
+
+The details of the syntax (actual names and symbols) are hard to pin down at
+this point, but we can talk about some of the concepts that we expect to come
+out in the syntax. For example, although Python, C++, and Java all have
+different surface syntax for constructing new objects, they all share the idea
+of the existence of classes and the existence of a function-like constructor
+that creates an instance of that class.
+
+The current image I have of the internal representation is that we need to be
+able to describe points and various types of point-sets (in particular lines
+and circles, maybe more). We need to be able to take intersections of point
+sets, choose points from point sets.
+
+I also think it would be cool (maybe even necessary) to be able to describe a
+construction as a sub-procedure. Perhaps we can define a construction which
+takes two points and gives you the midpoint of the segment between them.
+Perhaps the subprocedures can only take and return points? Perhaps they can
```

Comment:
Mhm. To continue our conversation I would say this.

A circle is an infinite set of points, but the position of all those points can be inferred if you know the location of the center, the location of another point, and have the [implicit] knowledge that you are dealing with a circle.